frontend/src/components/AddTrip/index.js

```diff
@@ -0,0 +1,112 @@
+      type="text"
+      placeholder={placeholder}
+      ref={ref}
+    />
+  );
+}
+
+function createFormGroup(controlId, formLabel, inputType, placeholder, ref) {
+  let formControl;
+  if (inputType === 'text') {
+    formControl = createTextFormControl(placeholder, ref)
+  } else {
+    // TODO(Issue #..): Create diff form control for destination, time,
+    //                  and collaborator emails
+    console.error('Only text form controls are implemented as of now');
+  }
+
+  return (
+    <Form.Group controlId={controlId}>
+      <Form.Label>{formLabel}</Form.Label>
+        {formControl}
+    </Form.Group>
+  )
+}
+
+/**
+ * Component corresponding to add trips modal.
+ *
+ * @extends React.Component
+ */
+class AddTrip extends React.Component {
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+
+    this.nameRef = React.createRef();
+    this.descriptionRef = React.createRef();
+    this.destinationRef = React.createRef();
+    this.startDateRef = React.createRef();
+    this.endDateRef = React.createRef();
+    this.collaboratorsRef = React.createRef();
+  }
+
+  /**
+   * Upon submission of form, creates a new Trip document, adds it to Firestore,
+   * and closes the add trip modal.
+   */
+  submitNewTrip = () => {
+    console.log('New trip submitted.');
+    createTrip(this.nameRef.current.value,
+               this.descriptionRef.current.value,
+               this.destinationRef.current.value,
+               this.startDateRef.current.value,
+               this.endDateRef.current.value,
+               this.collaboratorsRef.current.value
+              );
+
+    this.props.handleClose();
+  }
+
+  /** @inheritdoc */
+  render() {
+    return (
+      <Modal show={this.props.show} onHide={this.props.handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Add New Trip</Modal.Title>
+        </Modal.Header>
+
+        <Form>
+          <Modal.Body>
+            {createFormGroup('tripNameGroup', 'Trip Name', 'text',
+                             'Enter Trip Name', this.nameRef)}
+            {createFormGroup('tripDescripGroup', 'Trip Description', 'text',
+                             'Enter Trip Description', this.descriptionRef)}
+            {createFormGroup('tripDestGroup', 'Trip Destination', 'text',
+                             'Enter Trip Destination', this.destinationRef)}
+            {createFormGroup('tripStartDateGroup', 'Start Date', 'text',
+                             'Enter Trip Start Date', this.startDateRef)}
+            {createFormGroup('tripEndDateGroup', 'End Date', 'text',
+                             'Enter Trip End Date', this.endDateRef)}
+            {createFormGroup('tripCollabsGroup', 'Trip Collaborators', 'text',
+                           'Enter Collaborator Emails', this.collaboratorsRef)}
+            <Form.Text className="text-muted">
+              Enter emails in the form: 'user1@email.com, ..., userx@email.com'
+            </Form.Text>
```

Comment:
Are you going to do some crazy string operations for this? Could you have an "add collaborator" button and use Form.Control require and validate email address?

Reply:
This is related to the TODO I had above (Issue ..). Once I make different Form.Controls, it will probably be a system similar to what you mentioned where it would populate a new Form.Control with a new ref for an additional collaborator.

I should also probably validate the emails too so I will create an issue for that as well. Thanks for the reminder!